### PR TITLE
Update README.md

### LINKED ISSUE #473: Compiled Binaries in Release Folder
During the build process, the compiled signed binaries and `released` content should be moved to a root folder called `Release` or something along those lines to help folks know what is a valid assembly.

### PR DESCRIPTION
**Pull request by nr-ahemsath**
### Description

Resolves #473 by adding links to the download site and NuGet.org to the top-level README.md.

### Testing

N/A

### Changelog

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,8 @@ The New Relic Lambda Tracer is an OpenTracing Tracer implementation specifically
 
 ## Installation
 
-See our [agent installation documentation](https://docs.newrelic.com/docs/agents/net-agent/installation/introduction-net-agent-install).
+<!-- markdown-link-check-disable-next-line -->
+The latest released binaries for the .NET Agent can be found either [here](http://download.newrelic.com/dot_net_agent/latest_release/) or on [NuGet.org](https://www.nuget.org/profiles/newrelic).  See our [agent installation documentation](https://docs.newrelic.com/docs/agents/net-agent/installation/introduction-net-agent-install) to understand which of these files are appropriate for your application type and hosting environment.
```
Comment (angelatan2) [inline README.md:14]: Could you make sure that the sentence "See our ..." on the next line.?  I cannot tell from the diff comparison if it is continuing directly from the URL of Nuget.org.

**Review by angelatan2 — APPROVED** [review]
See comment.

File: README.md (hunk 1 above)
Comment (nr-ahemsath) [reply README.md]: Good idea, done.

**Comment by icpenguins** [comment]
@nr-ahemsath thank you! 👍

**Pull request merged**